src/transformers/models/bloom/modeling_bloom.py

```diff
@@ -89,6 +90,12 @@ def attention_mask_func(attention_scores, attention_mask, causal_mask):
         + ~causal_mask[:, :, key_length - query_length : key_length, :key_length]
     ).bool()
     padded_causal_mask = padded_causal_mask + attention_mask_bool[:, None, None, :key_length].bool()
+    
+    # ensure the initial prefix_length tokens attend to themselves
+    if prefix_length is not None:
+        # print(padded_causal_mask.shape)
+        padded_causal_mask[:,:,:prefix_length,:prefix_length] = True
```

Comment:
I don't know if actually it'd make a lot more sense to pass an actual mask? a bool tensor [batch_size, seq_length, seq_length]

Reply:
We could add as a `forward` argument in `prefix_len` in `BloomForPrefixLM` which is then converted to a mask. 


That mask can the be inserted to this function.
https://github.com/huggingface/transformers/blob/d0acc9537829e7d067edbb791473bbceb2ecf056/src/transformers/models/bloom/modeling_bloom.py#L287-L310

with slight changes `def forward(self, input, mask, max_positions, causal_mask=None):`

```
     if mask is not None: 
	         mask = mask.to(input.device) 
             if causal_mask is None:
        	         causal_mask = ( 
        	             torch.tril(torch.ones((max_positions, max_positions), dtype=torch.bool)) 
        	             .view(1, 1, max_positions, max_positions) 
        	             .to(input.device) 
        	         ) 
	         mask_output, padded_causal_mask = self.mask_func(input, mask, causal_mask) 
	         probs = nn.functional.softmax(mask_output, dim=-1, dtype=softmax_dtype) * (~padded_causal_mask) 
```